pkg/pagerduty/service.go

```diff
@@ -322,14 +326,34 @@ func (c *SvcClient) CreateService(data *Data) (string, error) {
 	}
 	data.ServiceID = newSvc.ID
 
-	data.IntegrationID, err = c.createIntegration(newSvc.ID, "V4 Alertmanager", "events_api_v2_inbound_integration")
-	if err != nil {
-		return "", fmt.Errorf("unable to create integration for service %v: %w", newSvc.ID, err)
+	var existingIntegration *pdApi.Integration
+
+	for k := range newSvc.Integrations {
+		integration := &newSvc.Integrations[k]
+
+		// The _reference suffix is added to the expected type as we are not manipulating a whole integration object there.
+		// See: https://developer.pagerduty.com/api-reference/165ad96a22ffd-get-a-service
+		if integration.Type == integrationType+"_reference" {
```

Comment:
Can we make `integrationType+"_reference"` a new variable to make it more descriptive? We'd also not need to re-construct it in a loop, and it makes the core loop logic easier to read. 
We can also use the ID instead of the pointer.

What do you think about something like this:

```
	// The _reference suffix is added to the expected type as we are not manipulating a whole integration object there.
	// See: https://developer.pagerduty.com/api-reference/165ad96a22ffd-get-a-service
	expectedIntegrationType := integrationType + "_reference"

	var existingIntegrationID string
	for _, integration := range newSvc.Integrations {
		if integration.Type == expectedIntegrationType {
			if existingIntegrationID == "" {
				existingIntegrationID = integration.ID
			} else {
				log.Log.WithValues("data.ServiceID", data.ServiceID, "integrationType", integrationType, "integration.ID", integration.ID).Info("service already has multiple integrations with the expected type, using first result")
			}
		}
	}

	if existingIntegrationID == "" {
		data.IntegrationID, err = c.createIntegration(newSvc.ID, integrationName, integrationType)
		if err != nil {
			return "", fmt.Errorf("unable to create integration for service %v: %w", newSvc.ID, err)
		}
	} else {
		data.IntegrationID = existingIntegrationID
	}
```